rwengine/src/script/ScriptModule.hpp

```diff
@@ -5,34 +5,215 @@
 
 #include <string>
 
+namespace script_bind
+{
+	template <class T>
+	struct arg_traits
+	{
+		static constexpr size_t arg_size = 1;
+	};
+
+	template <>
+	struct arg_traits<ScriptVec2>
+	{
+		static constexpr size_t arg_size = 2;
+	};
+
+	template <>
+	struct arg_traits<ScriptVec3>
+	{
+		static constexpr size_t arg_size = 3;
+	};
+
+	template <>
+	struct arg_traits<ScriptRGB>
+	{
+		static constexpr size_t arg_size = 3;
+	};
+
+	template <>
+	struct arg_traits<ScriptRGBA>
+	{
+		static constexpr size_t arg_size = 4;
+	};
+
+	template <class T, unsigned arg>
+	struct unpack
+	{
+		static T convert(const ScriptArguments& args)
+		{
+			return args.getParameter<T>(arg);
+		}
+	};
+
+	template <unsigned arg>
+	struct unpack<const ScriptArguments&, arg>
+	{
+		static const ScriptArguments& convert(const ScriptArguments& args)
+		{
+			return args;
+		}
+	};
+
+	template <class T, unsigned arg>
+	struct unpack<T&, arg>
+	{
+		static T& convert(const ScriptArguments& args)
+		{
+			return args.getParameterRef<T>(arg);
+		}
+	};
+
+	// unpack ScriptObjectType<T>& as non-ref so object is copied
+	// by the caller.
```

Comment:
was this comment break added by clang-format? there is a longer line at line 110 which is not broken?!